File: java-selenium/src/test/java/pages/LoginPage.java
package pages;

import config.Config;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage extends BasePage {


        private By usernameField = By.cssSelector("[name='UserName']");
        private By passwordField = By.cssSelector("[name='Password']");
        private By loginButton = By.id("login");
        private By statusMessage = By.id("loginstatus");

        public LoginPage(WebDriver driver) {
            super(driver);
        }

        public void open (){
            goTo(Config.BASE_URL + "/sampleapp");
        }

        public void enterCredentials(String username, String password){
            driver.findElement(usernameField).sendKeys(username);
            driver.findElement(passwordField).sendKeys(password);
        }

        public void clickLogin(){
            driver.findElement(loginButton).click();
        }

        public  void  login(String username, String password){
            open();
            enterCredentials(username, password);
            clickLogin();
        }

        public String getStatus(){
            return driver.findElement(statusMessage).getText();
        }

}

File: python-playwright/pages/login_page.py
from pages.base_page import BasePage
from playwright.sync_api import Page
from config import BASE_URL

class LoginPage(BasePage):

    def __init__(self, page: Page):
        super().__init__(page)

        self.username_field = "[name='UserName']"
        self.password_field = "[name='Password']"
        self.login_button = "#login"
        self.status_message = "#loginstatus"

    def login(self, username: str, password: str):
        self.go_to(f"{BASE_URL}/sampleapp")
        self.page.fill(self.username_field, username)
        self.page.fill(self.password_field, password)
        self.page.click(self.login_button)

    def get_status(self) -> str:
        return self.page.locator(self.status_message).inner_text()